core/mondoo-okta-security.mql.yaml

```
@@ -0,0 +1,632 @@
+
+        Run the following query:
+
+          ```mql
+          cnspec run okta -c "okta.applications.all ( _.signOnMode == "OPENID_CONNECT" ||  _.signOnMode == "SAML_2_0" )" --organization <org-name> --token <OKTA_TOKEN>
+          ```
+
+        __cnspec shell__
+
+        To audit Okta with `cnspec shell`:
+
+        1. Launch `cnspec shell`:
+
+          ```bash
+          cnspec shell okta ---organization <org-name> --token <OKTA_TOKEN>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          okta.applications.all ( _.signOnMode == "OPENID_CONNECT" ||  _.signOnMode == "SAML_2_0" )
+          ```
+      remediation: |
+        ### Okta Console
+
+        To update via the Okta portal:
+        1. Log in as Administrator to the related Okta organization (Identity Engine)
+        2. Select from Applications the "Applications" 
+        3. Select the "Create App Integration" or "Browse App Catalog" to see the available apps.
+        4. At the end the Users need to be assigned to the App. 
+    query: |
+      okta.applications.all ( _.signOnMode == "OPENID_CONNECT" ||  _.signOnMode == "SAML_2_0" )
+  - uid: mondoo-okta-security-okta-users-unlock
+    title: Ensure normal Users cannot Unlock themselves
+    severity: 60
+    docs:
+      desc: |
+
+        User account that has been locked due to exceeding the password entry limitation, or other reasons, should only inform the administrator and he/she should not be able to unlock themselves.
+
+        In contrast the self-service password recovery might be considered for Okta Admins, https://support.okta.com/help/s/article/How-to-configure-selfservice-account-recovery?language=en_US
+        
+      audit: |
+        __cnspec run__
+
+        To audit Okta `cnspec run`:
+
+        Run the following query:
+
+          ```mql
+          cnspec run okta -c "okta.policies.password.all( _.rules { actions["selfServiceUnlock"]["access"] == "DENY" })" --organization <org-name> --token <OKTA_TOKEN>
+          ```
+
+        __cnspec shell__
+
+        To audit Okta with `cnspec shell`:
+
+        1. Launch `cnspec shell`:
+
+          ```bash
+          cnspec shell okta ---organization <org-name> --token <OKTA_TOKEN>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          okta.policies.password.all( _.rules { actions["selfServiceUnlock"]["access"] == "DENY" }) 
+          ```
+      remediation: |
+        ### Okta Console
+
+        To update via the Okta portal:
+        1. Log in as Administrator to the related Okta organization (Identity Engine)
+        2. Select from Applications the "Applications" 
+        3. Select from Security the "Authenticators" and the password
+        4. You can add completely the new password policy by "Add New Password Policy" or edit the existent one. 
+        5. then "Add rule" and make sure under "Users can perform self-service", the "Unlock account" has been not selected. 
```

```suggestion
        5. Then "Add rule" and make sure under "Users can perform self-service", the "Unlock account" has been not selected. 
```

This sentence feels like it needs a rewrite. Are we having them select this?